### ISSUE TITLE
Gibiansky style and lens operators

### ISSUE THREAD

**Issue by pharpend** [chrisdone/hindent#37]
@gibiansky , I don't think this is what you imagined when it comes to lens operators?

```haskell
runStuff :: T.Text -> IO ()
runStuff txt = T.lines txt ~| drop 2 ~. T.splitOn "\t" ~. take 3 ~| foldl getContactInfo [] ~. T.intercalate
                                                                                                 ";" ~| T.intercalate
                                                                                                          "\n" ~| encodeUtf8 ~| Bs.hPut
                                                                                                                                  stdout
```

Note that these are my own lens-like operators, they don't come from Ed Kmett's library.

```haskell
(~.) :: Functor f => f a -> (a -> b) -> f b
(~.) = flip fmap

(~|) :: a -> (a -> b) -> b
(~|) = flip ($)
```

I guess this is yet another issue of operator precedence?

**Comment by gibiansky** [comment]
Here is a simpler example:
```haskell
asdfadsfadsf = aaaaaaaaaaaaaaaaaaa ~. aaaaaaaaaaaaaaaaa ~. aaaaaaaaaaaaaaaaaaaaaaa ~. aaaaaaaaaaaaaaaaaaaaaa ~. aaaaaaaaaaaaaaaa + (\x -> x + 1)
```
which currently yields
```haskell
asdfadsfadsf = aaaaaaaaaaaaaaaaaaa ~. aaaaaaaaaaaaaaaaa ~. aaaaaaaaaaaaaaaaaaaaaaa ~. aaaaaaaaaaaaaaaaaaaaaa ~. aaaaaaaaaaaaaaaa + (\x ->
                                                                                                                                      x + 1)
```

Effectively, operators are currently exempt from the line length rule, and thus it only kicks in for things like lambdas and multi-argument functions; at that point the baseline indent is very high and so we get the results we see here.

This should be fixed; am going to maybe work on it soonish.

**Comment by gibiansky** [comment]
I do not know how this could be done well, because operator formatting tends to be very variable.

As a result, I think I will allow the user to space operators themselves, and make `gibiansky` style respect user spacing to some extent – newlines will be kept.

**Comment by gibiansky** [comment]
Once #70 is merged, this can be closed. I do not think there is a general heuristic for spacing operators, both due to operator precedence issues and due to a variety of ways to space things in different places. I think leaving this as user-controlled is preferable to a variety of incomplete heuristics.
